src/time_integration/paired_explicit_runge_kutta/methods_PERK4.jl

```diff
@@ -0,0 +1,326 @@
+    f::F
+    alg::Alg # This is our own class written above; Abbreviation for ALGorithm
+    opts::PairedExplicitRKOptions
+    finalstep::Bool # added for convenience
+    dtchangeable::Bool
+    force_stepfail::Bool
+    # Additional PERK register
+    k1::uType
+end
+
+function init(ode::ODEProblem, alg::PairedExplicitRK4;
+              dt, callback::Union{CallbackSet, Nothing} = nothing, kwargs...)
+    u0 = copy(ode.u0)
+    du = zero(u0)
+    u_tmp = zero(u0)
+
+    k1 = zero(u0) # Additional PERK register
+
+    t0 = first(ode.tspan)
+    tdir = sign(ode.tspan[end] - ode.tspan[1])
+    iter = 0
+
+    integrator = PairedExplicitRK4Integrator(u0, du, u_tmp, t0, tdir, dt, dt, iter,
+                                             ode.p,
+                                             (prob = ode,), ode.f, alg,
+                                             PairedExplicitRKOptions(callback,
+                                                                     ode.tspan;
+                                                                     kwargs...),
+                                             false, true, false,
+                                             k1)
+
+    # initialize callbacks
+    if callback isa CallbackSet
+        for cb in callback.continuous_callbacks
+            throw(ArgumentError("Continuous callbacks are unsupported with paired explicit Runge-Kutta methods."))
+        end
+        for cb in callback.discrete_callbacks
+            cb.initialize(cb, integrator.u, integrator.t, integrator)
+        end
+    end
+
+    return integrator
+end
+
+@inline function last_three_stages!(integrator::PairedExplicitRK4Integrator, p, alg)
+    for stage in 1:2
+        @threaded for u_ind in eachindex(integrator.u)
+            integrator.u_tmp[u_ind] = integrator.u[u_ind] +
+                                      integrator.dt *
+                                      (alg.a_matrix_constant[1, stage] *
+                                       integrator.k1[u_ind] +
+                                       alg.a_matrix_constant[2, stage] *
+                                       integrator.du[u_ind])
+        end
+
+        integrator.f(integrator.du, integrator.u_tmp, p,
+                     integrator.t +
+                     alg.c[alg.num_stages - 3 + stage] * integrator.dt)
+    end
+
+    # Last stage
+    @threaded for i in eachindex(integrator.u)
+        integrator.u_tmp[i] = integrator.u[i] +
+                              integrator.dt *
+                              (alg.a_matrix_constant[1, 3] * integrator.k1[i] +
+                               alg.a_matrix_constant[2, 3] * integrator.du[i])
+    end
+
+    # Store K_{S-1} in `k1`:
+    @threaded for i in eachindex(integrator.u)
+        integrator.k1[i] = integrator.du[i]
+    end
+
+    integrator.f(integrator.du, integrator.u_tmp, p,
+                 integrator.t + alg.c[alg.num_stages] * integrator.dt)
+
+    @threaded for u_ind in eachindex(integrator.u)
```

Comment:
```suggestion
    @threaded for i in eachindex(integrator.u)
```
Maybe change this for consistency with other loops in.